src/main/kotlin/pl/edu/agh/gem/internal/solver/settlements/DebtRoundingPairingSolver.kt

```diff
@@ -0,0 +1,36 @@
+package pl.edu.agh.gem.internal.solver.settlements
+
+import pl.edu.agh.gem.internal.model.finance.balance.Balance
+import pl.edu.agh.gem.internal.model.finance.settelment.Settlement
+import java.math.BigDecimal
+
+object DebtRoundingPairingSolver {
+    fun solve(userBalances: List<Balance>): List<Settlement> {
+        val users = userBalances.map { it.copy() }
+        val creditors = users.filter { it.value > BigDecimal.ZERO }.sortedByDescending { it.value }.toMutableList()
+        val debtors = users.filter { it.value < BigDecimal.ZERO }.sortedByDescending { -it.value }.toMutableList()
+        val settlements = mutableListOf<Settlement>()
+
+        while (creditors.isNotEmpty() && debtors.isNotEmpty()) {
+            val creditor = creditors.removeAt(0)
+            val debtor = debtors.removeAt(0)
+            val amount = minOf(creditor.value, -debtor.value)
+
+            settlements.add(Settlement(debtor.userId, creditor.userId, amount))
+
+            val newCreditorBalance = creditor.value - amount
+            val newDebtorBalance = debtor.value + amount
+
+            if (newCreditorBalance > BigDecimal.ZERO) {
```

Comment:
we want to use compare() in that solver

Reply:
why? compairing by "<" and ">" for bigDecimals works fine (only == or != doesnt work)

Reply:
soo what will happen when we compare the same value but with difrent decimal point ?